### PR TITLE
feat(notificationStore): add delete notification action (#629)

### PR DESCRIPTION
**Pull request by swapnil-turing**

### CODE REVIEW

File: augment-store/server/authentication/tests.py [hunk 1]
```diff
@@ -176,6 +178,14 @@ def test_forgot_password(self):
         # AND the response should contain success message
         self.assertEqual(response.data["message"], "Password reset email sent")
 
+    def test_register_is_throttled(self):
+        url = reverse("v1:register")
+        payload = {"email": "limit@example.com", "password": "testpassword", "first_name": "Test", "last_name": "User"}
+        for _ in range(5):
+            self.client.post(url, payload)
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/authentication/tests.py]: value:useful; category:bug; feedback:Marked useful because this comment caught a real weakness in the new regression test before merge. It identified that duplicate registration payloads and missing success assertions could let the test pass without proving the throttle boundary was correct.

File: augment-store/server/authentication/tests.py [hunk 2]
```diff
@@ -3,13 +3,15 @@
...
 from django.test import override_settings
+from django.core.cache import cache
 from accounts.models import User
 
 
 class AuthenticationTests(BaseAPITestCase):
 
     def setUp(self):
         super().setUp()
+        cache.clear()
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/authentication/tests.py]: value:good-to-have; category:bug; feedback:Marked good-to-have because this comment improved the safety of the test harness even though it was not a production behavior bug. It pointed out that clearing the shared cache in every setup could leak interference into unrelated tests and create flaky results under parallel execution.

File: augment-store/server/authentication/views.py [hunk 3]
```diff
@@ -2,13 +2,25 @@
...
+
+class AuthUserThrottle(UserRateThrottle):
+    scope = "auth_user"
+    rate = "100/min"
+
+
 class RegisterView(CreateAPIView):
     serializer_class = RegisterSerializer
     permission_classes = []
+    throttle_classes = [AuthAnonThrottle, AuthUserThrottle]
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/authentication/views.py:23]: value:incorrect-but-reasonable; category:bug; feedback:Marked incorrect-but-reasonable because the behavior described is true but it does not indicate a defect in the intended design of this change. It highlights that anonymous registration is governed by the anonymous throttle alone, which matches the protection model used for this endpoint.

File: augment-store/server/authentication/views.py [hunk 4]
```diff
@@ -2,13 +2,25 @@
...
 from rest_framework.views import APIView
 from rest_framework.permissions import IsAuthenticated
+from rest_framework.throttling import AnonRateThrottle, UserRateThrottle
 
 from .serializers import ForgotPasswordSerializer, LoginSerializer, RefreshTokenSerializer, RegisterSerializer
 
 
+class AuthAnonThrottle(AnonRateThrottle):
+    scope = "auth_anon"
+    rate = "5/min"
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/authentication/views.py:12]: value:valid-but-wont-fix; category:documentation; feedback:Marked valid-but-wont-fix because this is a real deployment consideration, but it sits outside the code changes in this PR. It points out that a dummy cache backend disables DRF throttling entirely, which matters operationally even though the application code remains unchanged.

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/authentication/views.py [hunk 5]
```diff
@@ -2,13 +2,25 @@
 from rest_framework.generics import CreateAPIView
 from rest_framework.views import APIView
 from rest_framework.permissions import IsAuthenticated
+from rest_framework.throttling import AnonRateThrottle, UserRateThrottle
 
 from .serializers import ForgotPasswordSerializer, LoginSerializer, RefreshTokenSerializer, RegisterSerializer
 
 
+class AuthAnonThrottle(AnonRateThrottle):
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/authentication/views.py:10]: value:valid-but-wont-fix; category:documentation; feedback:Marked valid-but-wont-fix because this comment identified a real mismatch in the review context, but it does not require any code changes. It pointed out that stale PR metadata can mislead reviewers and release consumers about the actual behavior being merged.

**Comment by swapnil-turing** [comment]
augment review

**Review by michaelos443 — APPROVED** [review]
